src/lib/core/CHIPTLV.h

```diff
@@ -338,6 +359,21 @@ class DLL_EXPORT TLVReader
      */
     CHIP_ERROR Get(int16_t & v);
 
+    /**
+     * Get the value of the current element as a 24-bit signed integer.
+     *
+     * If the encoded integer value is larger than the output data type the resultant value will be
+     * truncated.
```

Comment:
I know the existing comments say that, but they are wrong: existing code returns error on invalid read attempts like that....